builtin/mainmenu/tab_online.lua

```diff
@@ -391,10 +397,37 @@ local function search_server_list(input)
 		return
 	end
 
+	local current_server = find_selected_server()
+
 	table.sort(search_result, function(a, b)
 		return a.points > b.points
 	end)
 	menudata.search_result = search_result
+
+	-- Keep current selection if it's in search results
+	local found_current = false
+	if current_server then
+	    for _, server in ipairs(search_result) do
+			if server.address == current_server.address and
+				server.port == current_server.port then
+				found_current = true
+				break
```

Comment:
Why not just `return` here (instead of `break`ing)? Then you can remove the `if not found_current` check further down which hopefully also saves some rebasing work once #15755 gets merged.